specutils/helcorr.py

```diff
@@ -0,0 +1,343 @@
+    dsinep = np.sin(deps)
+    dyahd = dcosep * dyhd - dsinep * dzhd
+    dzahd = dsinep * dyhd + dcosep * dzhd
+    dyabd = dcosep * dybd - dsinep * dzbd
+    dzabd = dsinep * dybd + dcosep * dzbd
+
+    dvelh = constants.au * (np.array([dxhd, dyahd, dzahd])) / u.second
+    dvelb = constants.au * (np.array([dxbd, dyabd, dzabd])) / u.second
+    return (dvelh, dvelb)
+
+
+# NOTE:
+# We may want to change the syntax input for helcorr so that it accepts a single
+# sky coordinate instead of ra/dec.
+# Similarly lon/lat/alt/jd could be replaced with a single astropy.units.Time
+# class.
+
+def helcorr(lon, lat, alt, ra, dec, mjd):
+    """
+    Calculate the heliocentric radial velocity corrections for an astronomical 
+    source.
+
+    Parameters
+    ----------
+    lon : `~astropy.coordinates.Longitude` or float
+        Earth longitude of the observatory (western direction is positive). Can
+        be anything that initialises an `~astropy.coordinates.Angle` object
+        (if float, in degrees).
+    lat : `~astropy.coordinates.Latitude` or float
+        Earth latitude of observatory. Can be anything that initialises an
+        `~astropy.coordinates.Latitude` object (if float, in degrees).
+    alt : `~astropy.units.Quantity` or float
+        Altitude of the observatory (if float, in meters).
+    ra : `~astropy.coordinates.Angle` or float
+        Right ascension of the object for epoch J2000 (if float, in degrees).
+    dec : `~astropy.coordinates.Angle` or float
+        Declination of the object for epoch J2000 (if float, in degrees).
+    mjd : float
+        The modified Julian date for the middle of exposure.
+
+    Returns
+    -------
+    helcorr : `~astropy.units.Quantity`
+        The heliocentric velocity correction.
+    """
+
+    if not isinstance(lon, coord.Longitude):
+        lon = coord.Longitude(lon * u.deg)
+
+    if not isinstance(lat, coord.Latitude):
+        lat = coord.Latitude(lat * u.deg)
+
+    if not isinstance(alt, u.Quantity):
+        alt *= u.m
+
+    if not isinstance(ra, u.Quantity):
+        ra *= u.deg
+
+    if not isinstance(dec, u.Quantity):
+        dec *= u.deg
+
+    # Here we specify the location so that we can easily calculate the mean
+    # local siderial time later on
+    time = astropy.time.Time(2.4e6 + mjd, format="jd", location=(lon, lat, alt))
+    epoch = time.datetime.year + time.datetime.month/12. \
+        + time.datetime.day/365.
+
+    # Precess the coordinates to the current epoch
+    coordinate = coord.SkyCoord(ra, dec, frame="fk5").transform_to(
+        coord.FK5(equinox="J{}".format(epoch)))
+
+    # Convert geodetic latitude into geocentric latitude to correct for rotation
+    # of the Earth
+    dlat = ((-11. * 60. + 32.743) * np.sin(2 * lat) + 1.1633 * np.sin(4 * lat) \
+        - 0.0026 * np.sin(6 * lat)) * u.degree
+    geocentric_lat = lat + dlat / 3600.
+
+    # Calculate distance of observer from Earth center
+    r = alt + 6378160.0 * u.m * (0.998327073 \
```

Comment:
Yes, the geoid is defined. See http://docs.astropy.org/en/latest/api/astropy.coordinates.EarthLocation.html#astropy.coordinates.EarthLocation
It can also passed on directly to `Time` if that is more helpful (sorry, have not looked at all at the code here, just answering @wkerzendorf's query).